### PR TITLE
[UnderScore] Update definitions of findIndex/findLastIndex

### PR DESCRIPTION
**Pull request by nallwhy**
As _predicates_ of `findIndex` / `findLastIndex` are transformed by `_.iteratee`, they can take _Object_ as a _predicates_.

And `findIndex` / `findLastIndex` can't return __Chain<T>_. I changed it to __ChainSingle<number>_.

[http://underscorejs.org/#iteratee](http://underscorejs.org/#iteratee)
[http://underscorejs.org/#findIndex](http://underscorejs.org/#findIndex)

### CODE REVIEW

**Comment by nallwhy** [comment]
As there are many other methods that transform their predicates through `_.iteratee`, I'm gonna check them later.

File: underscore/underscore.d.ts [hunk 1]
```diff
@@ -1017,6 +1002,11 @@ interface UnderscoreStatic {
 		predicate: _.ListIterator<T, boolean>,
 		context?: any): number;
 
+  findIndex<T, U extends {}>(
```
Comment (mhegazy) [inline underscore/underscore.d.ts]: `U extends {}` means the same thing as just `U`. it is rather confusing to use this for some declarations and not others. so i would recommend removing it.

File: underscore/underscore.d.ts [hunk 2]
```diff
@@ -1017,6 +1002,11 @@ interface UnderscoreStatic {
...
+    array: _.List<T>,
+    predicate: U,
```
Comment (mhegazy) [inline underscore/underscore.d.ts]: not sure i understand what this buys you here?  i would say it should just be

```ts
findIndex<T>(
 		array: _.List<T>,
  		predicate: _.ListIterator<T, boolean> | {},
  		context?: any): number;
```

Comment (mhegazy) [inline underscore/underscore.d.ts]: also indentation is all messed up through out the file. can you please fix these.

**Comment by nallwhy** [comment]
@mhegazy I fixed it. Thank you!

**Pull request merged**